app/components/billing-status-badge.ts

```diff
@@ -22,10 +22,48 @@ export default class BillingStatusDisplayComponent extends Component<Signature>
     return this.currentUser?.activeDiscountForYearlyPlan || null;
   }
 
+  get activeDiscountForYearlyPlanExcludingStage2CompletionDiscount(): PromotionalDiscountModel | null {
+    return this.currentUser?.activeDiscountForYearlyPlanExcludingStage2CompletionDiscount || null;
+  }
+
+  get badgeType(): 'vip' | 'member' | 'old-discounts' | 'all-discounts' | 'free-weeks-left' | 'upgrade' {
```

Comment:
I now see this this is required, ignore. Name still seems confusing (old vs. all), would do something like this:

```suggestion
  get badgeType(): 'vip' | 'member' | 'discount-timer' | 'discount-timer-excluding-stage-2-completion' | 'free-weeks-left' | 'upgrade' {
```